service/discovery/azure/compute.go

```diff
@@ -207,33 +213,26 @@ func (d *azureComputeDiscovery) discoverFunctionsWebApps() ([]voc.IsCloudResourc
 	return list, nil
 }
 
-func (d *azureComputeDiscovery) handleFunction(function *armappservice.Site) voc.IsCompute {
+func (d *azureComputeDiscovery) handleFunction(site *armappservice.Site, config armappservice.WebAppsClientGetConfigurationResponse) voc.IsCompute {
 	var (
-		runtimeLanguage     string
-		runtimeVersion      string
-		config              armappservice.WebAppsClientGetConfigurationResponse
-		err                 error
-		publicNetworkAccess = false
+		ni              []voc.ResourceID
+		publicAccess    bool
+		resourceLogging *voc.ResourceLogging
+		runtimeLanguage string
+		runtimeVersion  string
 	)
 
-	// If a mandatory field is empty, the whole function is empty
-	if function == nil {
+	if site == nil || config == (armappservice.WebAppsClientGetConfigurationResponse{}) {
+		log.Error("input parameter empty")
 		return nil
 	}
 
-	if util.Deref(function.Properties.PublicNetworkAccess) == "Enabled" {
-		publicNetworkAccess = true
-	}
-
-	if *function.Kind == "functionapp,linux" { // Linux function
-		runtimeLanguage, runtimeVersion = runtimeInfo(*function.Properties.SiteConfig.LinuxFxVersion)
-	} else if *function.Kind == "functionapp" { // Windows function, we need to get also the config information
-		// Get site config
-		config, err = d.clients.sitesClient.GetConfiguration(context.Background(), *function.Properties.ResourceGroup, *function.Name, &armappservice.WebAppsClientGetConfigurationOptions{})
-		if err != nil {
-			log.Errorf("error getting site config: %v", err)
-		}
+	// Get properties for the function
+	ni, publicAccess, resourceLogging = d.getSiteProperties(site)
 
+	if *site.Kind == "functionapp,linux" { // Linux function
```

Comment:
Suggestion: Maybe extract thi big if and else to a separate function to keep the code in this method cleaner. Personal preference though